wasm-chain-integration/src/v1/mod.rs

```diff
@@ -0,0 +1,853 @@
+            // - negative values lead to a rejection with a specific reject reason.
+            if let Some(Value::I32(n)) = result {
+                if n == 0 {
+                    Ok(InitResult::Success {
+                        state: host.state,
+                        logs: host.logs,
+                        return_value: host.return_value,
+                        remaining_energy,
+                    })
+                } else {
+                    Ok(InitResult::Reject {
+                        reason: reason_from_wasm_error_code(n)?,
+                        return_value: host.return_value,
+                        remaining_energy,
+                    })
+                }
+            } else {
+                bail!("Wasm module should return a value.")
+            }
+        }
+        Ok(ExecutionOutcome::Interrupted {
+            reason,
+            config: _,
+        }) => match reason {},
+        Err(e) => {
+            if e.downcast_ref::<OutOfEnergy>().is_some() {
+                Ok(InitResult::OutOfEnergy)
+            } else {
+                Err(e)
+            }
+        }
+    }
+}
+
+/// Response from an invoke call.
+pub enum InvokeResponse {
+    /// Execution was successful, and the state potentially changed.
+    Success {
+        /// New state, if it changed.
+        new_state:   Option<v0::State>,
+        /// Balance after the execution of the interrupt.
+        new_balance: Amount,
+        /// Some calls do not have any return values, such as transfers.
+        data:        Option<ParameterVec>,
+    },
+    /// Execution was not successful. The state did not change
+    /// and the contract responded with the given error code and data.
+    Failure {
+        code: u64,
+        data: Option<ParameterVec>,
+    },
+}
+
+/// Invokes an init-function from a given artifact *bytes*
+#[cfg_attr(not(feature = "fuzz-coverage"), inline)]
+pub fn invoke_init_from_artifact<'a>(
+    artifact_bytes: &'a [u8],
+    amount: u64,
+    init_ctx: impl v0::HasInitContext,
+    init_name: &str,
+    parameter: ParameterRef,
+    energy: InterpreterEnergy,
+) -> ExecResult<InitResult> {
+    let artifact = utils::parse_artifact(artifact_bytes)?;
+    invoke_init(artifact, amount, init_ctx, init_name, parameter, energy)
+}
+
+/// Invokes an init-function from Wasm module bytes
+#[cfg_attr(not(feature = "fuzz-coverage"), inline)]
+pub fn invoke_init_from_source(
+    source_bytes: &[u8],
+    amount: u64,
+    init_ctx: impl v0::HasInitContext,
+    init_name: &str,
+    parameter: ParameterRef,
+    energy: InterpreterEnergy,
+) -> ExecResult<InitResult> {
+    let artifact = utils::instantiate(&ConcordiumAllowedImports, source_bytes)?;
+    invoke_init(artifact, amount, init_ctx, init_name, parameter, energy)
+}
+
+/// Same as `invoke_init_from_source`, except that the module has cost
+/// accounting instructions inserted before the init function is called.
+/// metering.
```

Comment:
I suppose the _metering_ should replace _cost accounting instructions_ or simply be removed.

Reply:
Should be removed.